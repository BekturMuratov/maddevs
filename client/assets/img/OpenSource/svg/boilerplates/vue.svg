<svg xmlns="http://www.w3.org/2000/svg" width="336" height="125" fill="none" xmlns:v="https://vecta.io/nano"><g clip-path="url(#A)"><path d="M206.784.001l-38.797 67.203L129.191.001H-.008l167.996 290.997L335.984.001h-129.2z" fill="#41b883"/><path d="M206.785.001l-38.797 67.203L129.191.001h-62l100.798 174.595L268.786.001h-62.001z" fill="#34495e"/></g><defs><clipPath id="A"><path fill="#fff" d="M0 0h336v291H0z"/></clipPath></defs></svg>