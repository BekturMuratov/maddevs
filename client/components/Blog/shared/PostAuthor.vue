<template>
  <div class="blog-post__author">
    <img
      v-if="document.author_image.url !== undefined"
      :data-src="document.author_image.url"
      :alt="$prismic.asText(document.author)"
      ref="authorImage"
      class="blog-post__author-image img_lazy"
    >
    <div
      v-else
      class="blog-post__none-image"
    />
    <div class="blog-post__author-info">
      <p class="blog-post__author-name">
        {{ shortTitle }}
      </p>
      <div class="blog-post__data-of-creation">
        <span class="blog-post__author-title">{{ document.author_title }}</span>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'PostAuthor',
  props: {
    document: {
      type: Object,
      required: true,
    },
  },

  computed: {
    shortTitle() {
      return this.$prismic.asText(this.document.author).substr(0, 100)
    },
  },

  watch: {
    document() {
      this.$refs.authorImage.classList.remove('img_lazy-fade')
      this.$refs.authorImage.classList.add('img_lazy')
    },
  },
}
</script>

<style scoped lang="scss">
@import '../../../assets/styles/vars';
.blog-post {
  &__author {
    display: flex;
    align-items: center;
  }

  &__author-info {
    margin-left: 9px;
  }

  &__author-name {
    color: $text-color--white;
  }

  &__author-title,
  &__author-name {
    display: block;
    font-size: 13px;
    font-family: 'Inter', sans-serif;
    font-weight: 400;
    line-height: 129%;
    letter-spacing: -0.02em;
  }

  &__author-name {
    margin-bottom: 4px;
  }

  &__author-image {
    object-fit: cover;
  }

  &__author-image,
  &__none-image {
    width: 36px;
    height: 36px;
    border-radius: 100%;
  }

  &__none-image {
    background-color: $bgcolor--black-light;
  }

  &__author-title {
    color: $text-color--grey-pale;
  }
}
</style>
