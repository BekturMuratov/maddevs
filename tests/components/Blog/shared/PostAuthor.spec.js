import { render } from '@testing-library/vue'
import PostAuthor from '@/components/Blog/shared/PostAuthor'
import { shallowMount } from '@vue/test-utils'

const mocks = {
  $prismic: {
    asText: text => text,
  },
}

const props = {
  document: {
    author: 'some-author',
    author_image: {
      url: 'some-url',
    },
    author_title: 'author_title',
  },
}

describe('text slice component', () => {
  it('should render correctly', () => {
    const { container } = render(PostAuthor, {
      props,
      mocks,
    })

    expect(container).toMatchSnapshot()
  })

  it('should correct work watcher', () => {
    const remove = jest.fn()
    const add = jest.fn()
    const callObject = {
      $refs: {
        authorImage: {
          classList: {
            remove,
            add,
          },
        },
      },
    }
    const wrapper = shallowMount(PostAuthor, {
      propsData: props,
      mocks,
    })

    wrapper.vm.$options.watch.document.call(callObject)

    expect(remove).toHaveBeenCalledTimes(1)
    expect(add).toHaveBeenCalledTimes(1)
  })
})
