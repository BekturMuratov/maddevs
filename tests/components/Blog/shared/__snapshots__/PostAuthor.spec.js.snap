// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text slice component is a Vue instance 1`] = `
<div>
  <div
    class="blog-post__author"
  >
    <img
      alt="some-author"
      class="blog-post__author-image img_lazy"
      data-src="some-url"
    />
     
    <div
      class="blog-post__author-info"
    >
      <p
        class="blog-post__author-name"
      >
        
      some-author
    
      </p>
       
      <div
        class="blog-post__data-of-creation"
      >
        <span
          class="blog-post__author-title"
        >
          author_title
        </span>
      </div>
    </div>
  </div>
</div>
`;
